system-addon/lib/TopSitesFeed.jsm

```diff
@@ -100,7 +106,7 @@ this.TopSitesFeed = class TopSitesFeed {
         {isDefault: !!notBlockedDefaultSites.find(finder)}, link, {hostname: shortURL(link)});
 
       // Add in favicons if we don't already have it
-      if (!copy.favicon) {
+      if (!copy.favicon && !copy.customScreenshotURL) {
```

Comment:
What is this guarding? This code path should be for pinned links inheriting properties from frecent links, but frecent links shouldn't have a custom screenshot anyway?

Reply:
You're right. This is done as a map over pinned items and I think I just got confused.